x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,209 @@
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+
+settings.monitoring:
+  use_output: monitoring
+
+datasources:
+  # use the nginx package
+  - id: nginx-x1
+    title: "This is a nice title for human"
+    # Package this config group is coming from. On importing, we know where it belongs
+    # The package tells the UI which application to link to
+    package:
+      name: epm/nginx
+      version: 1.7.0
+    namespace?: prod
+    constraints:
+      # Contraints look are not final
+      - os.platform: { in: "windows" }
+      - agent.version: { ">=": "8.0.0" }
+    inputs:
+      - type: logs
+        processors?:
+        streams:
+          - id?: {id}
+            dataset: nginx.acccess
+            paths: /var/log/nginx/access.log
+          - id?: {id}
+            dataset: nginx.error
+            paths: /var/log/nginx/error.log
+      - type: metrics/nginx
+        streams:
+          - id?: {id}
+            dataset: nginx.stub_status
+            metricset: stub_status
+
+  #################################################################################################
+  # Custom Kafka datasource
+  - id: kafka-x1
+    title: "Consume data from kafka"
+    namespace?: prod
+    use_output: long_term_storage
+    inputs:
+      - type: kafka
+        host: localhost:6566
+        streams:
+          - dataset: foo.dataset
+            topic: foo
+            processors:
+             - extract_bro_specifics
+
+
+  #################################################################################################
+  # System EPM package
+  - id: system
+    title: Collect system information and metrics
+    package:
+      name: epm/system
+      version: 1.7.0
+    inputs:
+      - type: metrics/system
+        streams:
+         - id?: {id}
+           enabled?: false # default true
+           metricset: cpu
+           period: 10s
+           dataset: system.cpu
+           metrics: ["percentages", "normalized_percentages"]
+         - metricset: memory
+           dataset: system.memory
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
```

Comment:
First, I  think we trying to optimize for the UI experience vs the YAML edition experience. 

If you look at the metricbeat module definition in beats they are do have settings, they are not just _opt-in_ thing. I've used the _system_ module because it's one of the bigger one and if you look at the metricset they do have settings.  I do not find that obvious to change these options using the key/name approache.

But @ruflin do you have a strong opinion? I think there were also a reason concerning `dataset` for using the list.

```

  cpu.metrics:  ["percentages"]  # The other available options are normalized_percentages and ticks.
  core.metrics: ["percentages"]  # The other available option is ticks.
  #filesystem.ignore_types: []
  #process.include_top_n:
    #enabled: true
    #by_cpu: 0
    #by_memory: 0
  #process.cmdline.cache.enabled: true
  #process.cgroups.enabled: true
  #process.env.whitelist: []
  #process.include_cpu_ticks: false
  #raid.mount_point: '/'

  #socket.reverse_lookup.enabled: false
  #socket.reverse_lookup.success_ttl: 60s
  #socket.reverse_lookup.failure_ttl: 60s
  #diskio.include_devices: []

Reply:
@roncohen  What do you think about metrics module options as defined above? Where they fit in the shortened version?

Reply:
OK, we can add more defaults and a rewrite step later 👍